### PR TITLE
Fixed call for select_related without arguments

### LINKED ISSUE #11: Select related is called without a value, creating n joins
According to Django docs:
> There may be some situations where you wish to call select_related() with a lot of related objects, or where you don’t know all of the relations. In these cases it is possible to call select_related() with no arguments. This will follow all non-null foreign keys it can find - nullable foreign keys must be specified. This is not recommended in most cases as it is likely to make the underlying query more complex, and return more data, than is actually needed.

https://docs.djangoproject.com/en/2.1/ref/models/querysets/#select-related

And an issue was found here (master):

https://github.com/tfoxy/graphene-django-optimizer/blob/a2522943016f58dd250f3f9ea6a88af539a94bbb/graphene_django_optimizer/query.py#L334

There is a call to select related with an empty list. this triggers `%` joins.

I first saw this when I debugged this, and so many fields were added, and I could not understand why.
The tests in this repo did not catch this because they are all `null=True`.

To reproduce:

```python
# Models
class SomeOtherItem(models.Model):
    name = models.CharField(max_length=100, blank=True)


class OtherItem(models.Model):
    name = models.CharField(max_length=100, blank=True)
    some_other_item = models.ForeignKey('SomeOtherItem', on_delete=models.PROTECT, null=False)

# Schema

class SomeOtherItemType(DjangoObjectType):
    class Meta:
        model = SomeOtherItem


class OtherItemType(DjangoObjectType):
    class Meta:
        model = OtherItem

class Query(graphene.ObjectType):
    other_items = graphene.List(OtherItemType)

    def resolve_other_items(root, info):
        return gql_optimizer.query(OtherItemType.objects.all(), info)

# Test
def test_should_only_use_the_only_and_not_select_related():
    info = create_resolve_info(schema, '''
        query {
            otherItems {
                id
                name
            }
        }
    ''')
    qs = OtherItem.objects.all()
    items = gql_optimizer.query(qs, info)
    optimized_items = qs.only('id', 'name')
    assert_query_equality(items, optimized_items)

```

### PR DESCRIPTION
**Pull request by yardensachs**
Hi,

This fixes #11

I added new models/types, because changing the existing ones made the other tests fail, and that was a lot of changing to do.

### CODE REVIEW

File: graphene_django_optimizer/query.py [hunk 1]
```diff
@@ -331,10 +331,15 @@ def abort_only_optimization(self):
         self.only_list = None
```
Comment (yardensachs) [inline graphene_django_optimizer/query.py]: This file change is the actual fix ↓

Comment (yardensachs) [inline graphene_django_optimizer/query.py]: This file change is the actual fix ↓

File: tests/test_query.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 from .graphql_utils import create_resolve_info
 from .models import (
-    Item,
+    Item, OtherItem
```
Comment (tfoxy) [inline tests/test_query.py]: Can you put `OtherItem` in another line with a comma? Like the following:

```py
from .models import (
    Item,
    OtherItem,
)
```

Comment (yardensachs) [reply tests/test_query.py]:
```suggestion
    Item,
    OtherItem,
```

**Pull request merged**